xbmc/settings/AdvancedSettings.cpp

```diff
@@ -821,6 +823,8 @@ void CAdvancedSettings::ParseSettingsFile(const std::string &file)
     XMLUtils::GetInt(pElement, "curlclienttimeout", m_curlconnecttimeout, 1, 1000);
     XMLUtils::GetInt(pElement, "curllowspeedtime", m_curllowspeedtime, 1, 1000);
     XMLUtils::GetInt(pElement, "curlretries", m_curlretries, 0, 10);
+    XMLUtils::GetInt(pElement, "curlkeepaliveinterval", m_curlKeepAliveInterval, 1, 300);
+    XMLUtils::GetBoolean(pElement, "curlkeepalive", m_curlKeepAlive);
```

Comment:
Perhaps drop "curlkeepalive" and just disable keep-alive when "curlkeepaliveinterval" is 0 ? Also drop m_curlKeepAlive then ofc.

Reply:
yes, good idea, done.
We could actually remove m_keepAliveInterval from CurlFile as well, since we always ask advancedsettings anyway. Same for a couple other existing member vars there.

Reply:
Yes, sounds good. Code will become cleaner as well (like I suggested in my other comment_